File: PluginMessagingUtils/src/main/java/org/openredstone/executors/RenameItemExecutor.java
package org.openredstone.executors;

import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.plugin.Plugin;
import org.openredstone.messaging.Message;

public class RenameItemExecutor extends Executor {

    public RenameItemExecutor(Plugin plugin) {
        super(plugin);
    }

    @Override
    public void execute(Message message) throws Exception {
        Player player = plugin.getServer().getPlayer(message.getUuid());

        if (player == null) {
            throw new Exception("Player not found.");
        }

        if (message.getArguments().length < 1) {
            throw new Exception("Not enough arguments");
        }

        ItemStack mainHand = player.getInventory().getItemInMainHand();

        if (mainHand.getType().equals(Material.AIR)) {
            throw new Exception("Names cannot be applied to item type in main hand.");
        }

        ItemMeta meta = mainHand.getItemMeta();
        meta.setDisplayName(message.getArguments()[0]);
        mainHand.setItemMeta(meta);

    }
}

File: PluginMessagingUtils/src/main/java/org/openredstone/executors/RenameItemExecutor.kt
package org.openredstone.executors

import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.plugin.Plugin
import org.openredstone.messages.ActionMessage

class RenameItemExecutor(plugin: Plugin) : Executor(plugin) {
    override fun execute(actionMessage: ActionMessage) {
        val player: Player =
            plugin.server.getPlayer(actionMessage.uuid)
                ?: throw Exception("Player not found.")
        if (actionMessage.arguments.isEmpty()) {
            throw Exception("Not enough arguments")
        }
        val mainHand = player.inventory.itemInMainHand
        if (mainHand.type == Material.AIR) {
            throw Exception("Names cannot be applied to item type in main hand.")
        }
        val meta = mainHand.itemMeta
        meta!!.setDisplayName(actionMessage.arguments[0])
        mainHand.itemMeta = meta
    }
}
